extensions/skins/conversation_v1/player.html

```diff
@@ -2,107 +2,258 @@
-      <md-card class="conversation-skin-exploration-intro-card">
-        <img ng-src="<[introCardImageUrl]>">
-        <div class="conversation-skin-exploration-header">
-          <h3 itemprop="description" role="heading" class="protractor-test-exploration-header"><[explorationTitle]></h3>
+  <!-- Off-screen preview of future card in order to pre-determine the target height for the animation. -->
+  <md-card class="conversation-skin-tutor-card" style="position: absolute; left: -30000px; top: -30000px;">
+    <div id="futurePreview">
+      <div class="conversation-skin-tutor-card-top-section">
+        <img class="conversation-skin-initial-oppia-avatar" src="/images/general/o_black_72px.png">
+        <div angular-html-bind="upcomingContentHtml">
         </div>
-      </md-card>
+      </div>
 
-      <div class="conversation-skin-iframe-container">
-        <oppia-gadget-panel panel-contents="gadgetPanelsContents.left" class="conversation-skin-left-gadget-panel"></oppia-gadget-panel>
-        <oppia-gadget-panel panel-contents="gadgetPanelsContents.right" class="conversation-skin-right-gadget-panel"></oppia-gadget-panel>
-        <div ng-repeat="state in allResponseStates track by $index"
-             ng-class="{'conversation-skin-last-log-entry': $last}">
-          <md-card class="conversation-skin-card">
-            <div class="conversation-skin-card-feedback-toggle-container" ng-if="state.stateName && !isInPreviewMode">
-              <span class="glyphicon glyphicon-comment conversation-skin-card-feedback-toggle" popover-placement="<[isWindowNarrow() ? 'left' : 'bottom']>" popover-template="popover/feedback" popover-trigger="click" tooltip="Feedback" tooltip-placement="bottom" state-name="<[state.stateName]>">
-              </span>
-            </div>
+      <div ng-if="activeCard.interactionIsInline && upcomingInlineInteractionHtml">
+        <div class="conversation-skin-inline-interaction">
+          <img class="conversation-skin-user-avatar img-circle" ng-src="<[profilePicture]>">
+          <div angular-html-bind="upcomingInlineInteractionHtml">
+          </div>
+        </div>
+      </div>
+    </div>
+  </md-card>
+
+  <!-- TODO(sll): do we need to preserve the itemprop attr? -->
+  <div role="main" style="margin: 0 auto; position: relative;">
+    <progress-dots num-dots="numProgressDots" current-dot-index="currentProgressDotIndex" class="conversation-skin-progress-dots">
+    </progress-dots>
+
+    <div class="conversation-skin-cards-container conversation-skin-animate-cards" ng-class="{'animate-to-two-cards': isAnimatingToTwoCards, 'animate-to-one-card': isAnimatingToOneCard}">
+      <md-card class="conversation-skin-tutor-card conversation-skin-tutor-card-active" ng-class="{'conversation-skin-left-card': !activeCard.interactionIsInline && canWindowFitTwoCards(), 'conversation-skin-has-subsidiary-cards': !activeCard.interactionIsInline && !canWindowFitTwoCards()}" ng-show="isPanelVisible(PANEL_TUTOR)">
```

Comment:
Might be worth adding a comment describing this complex behavior. From what I understand, there are three situations:

1. There is an inline interaction and therefore one card.
2. There is a supplemental interaction, requiring two cards. Both fit on one screen.
3. Same as 2, but the screen is too small to fit both cards. This requires a bar at the bottom of the screen to switch between the tutor and interaction panels.